test/holepuncher.js

```diff
@@ -0,0 +1,148 @@
+    [{ host: '192.168.122.238' }, { host: '172.17.0.1' }]
+  ), { host: '192.168.122.238' })
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.122.238' }, { host: '172.17.0.1' }],
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.17.0.1' }]
+  ), { host: '192.168.122.1' })
+
+  // (host with docker, vm with docker)
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.17.0.1' }],
+    [{ host: '192.168.122.238' }]
+  ), { host: '192.168.122.238' })
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.122.238' }],
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.17.0.1' }]
+  ), { host: '192.168.122.1' })
+})
+
+test('holepuncher match - host vs container', async function (t) {
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.17.0.1' }],
+    [{ host: '172.17.0.2' }]
+  ), { host: '172.17.0.2' })
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '172.17.0.2' }],
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.17.0.1' }]
+  ), { host: '172.17.0.1' })
+})
+
+test('holepuncher match - container vs container (on same host)', async function (t) {
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '172.17.0.3' }],
+    [{ host: '172.17.0.2' }]
+  ), { host: '172.17.0.2' })
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '172.17.0.2' }],
+    [{ host: '172.17.0.3' }]
+  ), { host: '172.17.0.3' })
+})
+
+test.skip('holepuncher match - container on host vs container on virtual machine', async function (t) {})
+
+test.skip('holepuncher match - host vs container on virtual machine', async function (t) {})
+test.skip('holepuncher match - container on host vs virtual machine', async function (t) {})
+
+test('holepuncher match - custom made', async function (t) {
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.122.238' }],
+    [{ host: '172.16.1.1' }, { host: '10.0.0.5' }]
+  ), { host: '172.16.1.1' })
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '10.1.2.3' }],
+    [{ host: '192.168.0.23' }, { host: '172.16.1.1' }, { host: '10.4.5.6' }, { host: '192.168.122.1' }]
+  ), { host: '10.4.5.6' }, 'Same network (first segment)')
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '10.0.2.3' }],
+    [{ host: '192.168.0.23' }, { host: '172.16.1.1' }, { host: '10.4.5.6' }, { host: '10.0.5.6' }, { host: '192.168.122.1' }]
+  ), { host: '10.0.5.6' }, 'Same network (second segment)')
+
+  t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.122.238' }],
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '172.16.1.1' }]
+  ), { host: '192.168.122.1' }, 'Same subnet (third segment)')
+
+  /* t.alike(Holepuncher.matchAddress(
+    [{ host: '192.168.122.238' }],
+    [{ host: '192.168.0.23' }, { host: '192.168.122.1' }, { host: '192.168.122.238' }, { host: '172.16.1.1' }]
+  ), { host: '192.168.122.238' }, 'Full match') */
```

Comment:
why is this commented out?

Reply:
It doesn't work because I invented that list of addresses, so I think it's invalid that it has this two:
`{ host: '192.168.122.1' }, { host: '192.168.122.238' }`

Currently it will pick up `192.168.122.1`, but I think only "routers" have the `.1`, right? For example, if my computer has a docker network, and a virtual machines network, then my same computer will have two networks that ends with `.1`

Reply:
makes sense that it picks the first one